### PR TITLE
Updated Binary LED Clock (BLC)

### PR DESCRIPTION
**Pull request by aemkai**
added settings page and functionality for customizing LED colors

### CODE REVIEW

File: apps/blc/blc.js [hunk 1]
```diff
@@ -2,135 +2,181 @@
...
+            col_off = black;
         }
-        
+        else
+        {
+            bord_col = black;   
+            col_off = white;
+        }
+
+        let pwr2 = [1, 2, 4, 8, 16, 32];        // array with powers of 2, because poweroperator '**' doesnt work
```
Comment (bobrippling) [inline apps/blc/blc.js:66]: Instead of this you can use bitshifting if you like:

```diff
-pwr2[bit_cnt]
+1 << bit_cnt
```

File: backup.js [hunk 2]
```diff
@@ -55,7 +55,7 @@ function bangleDownload() {
     showToast('Backup complete!', 'success');
     if (typeof Android !== "undefined" && typeof Android.saveFile === 'function') {
       // Recent Gadgetbridge version that provides the saveFile interface
-      Android.saveFile("Banglejs backup.zip", "application/zip", content);
+      Android.saveFile("Banglejs backup.zip", "application/zip", btoa(content));
```
Comment (bobrippling) [inline backup.js:58]: Does this fix a bug? Let's pull it out to a separate PR if so

**Review by bobrippling — COMMENTED** [review]
Thanks for the PR - had a few comments

File: apps/blc/blc.js (hunk 1 above)
Comment (aemkai) [reply apps/blc/blc.js:66]: Yes, this is also a good solution. In my opinion the the "LUT"-way may be a bit faster an with only six values not much bigger regarding code size.

File: backup.js (hunk 2 above)
Comment (aemkai) [reply backup.js:58]: I have no clue what this is? I didn't create a backup.js

Comment (bobrippling) [reply backup.js:58]: No problem - looks like it's a github thing, seems fine locally

File: apps/blc/blc.js (hunk 1 above)
Comment (bobrippling) [reply apps/blc/blc.js:66]: Sure, we can keep it as-is for now (the bit shift will be faster as there's no memory indirection or bounds checking)
